File: TwoPointers/SortedSquares.js
function SortedSquares(nums) {
    let left = 0;
    let right = nums.length - 1;
    const result = new Array(nums.length);

    for (let i = nums.length - 1; i >= 0; i--) {
        if (Math.abs(nums[left]) > Math.abs(nums[right])) {
            result[i] = Math.pow(nums[left], 2);
            left++;
        } else {
            result[i] = Math.pow(nums[right], 2);
            right--;
        }
    }

    return result;
}

console.log(SortedSquares([-4,-1,0,3,10]));

File: TwoPointers/sorted_squares.py
def sorted_squares(nums: list[int]) -> list[int]:
    n = lens(nums)
    left, right = 0, n - 1
    result = [0] * n

    for i in range(n, 0, -1):
        if abs(nums[left]) > abs(nums[right]):
            result[i] = nums[left] ** 2
            left += 1
        else:
            result[i] = nums[right] ** 2
            right -=1

    return result


print(sorted_squares([-4,-1,0,3,10]))